### PR TITLE
Fixes #9678 - Can't update admin flag for users via API

### PR DESCRIPTION
**Pull request by tstrachota**

### CODE REVIEW

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -1,9 +1,8 @@
 module Api
   module V1
     class UsersController < V1::BaseController
-      include Foreman::Controller::UsersMixin
-
       before_filter :find_resource, :only => %w{show update destroy}
+      include Foreman::Controller::UsersMixin
```
Comment (ares) [inline app/controllers/api/v1/users_controller.rb:6]: just reordering? or was there some specific reason? the order is different in v2 controller

File: app/controllers/api/v2/users_controller.rb [hunk 2]
```diff
@@ -8,7 +8,6 @@ class UsersController < V2::BaseController
       include Api::Version2
       include Api::TaxonomyScope
       before_filter :find_optional_nested_object
-      before_filter :find_resource, :only => [:show, :update, :destroy]
```
Comment (ares) [inline app/controllers/api/v2/users_controller.rb:11]: good catch, :dart:

File: app/controllers/api/v1/users_controller.rb (hunk 1 above)
Comment (ares) [inline app/controllers/api/v1/users_controller.rb:6]: ah now I get it, it's actual fix, as the mixin adds before filters, could you add # comment so we don't reorder it later?

Comment (tstrachota) [inline app/controllers/api/v1/users_controller.rb:6]: The test I created should catch possible reordering issues but I can add the comment too.

Comment (ares) [inline app/controllers/api/v1/users_controller.rb:6]: please add it, as the failing test in this case won't make the cause clear I think

**Comment by tstrachota** [comment]
[test]

**Comment by ares** [comment]
ready to merge pending tests

**Comment by ares** [comment]
Merged as https://github.com/theforeman/foreman/commit/1b1b39861e485523b0cc0c6435fef30c38df7e07, thanks @tstrachota!

**Pull request closed without merging**